src/entries/popup/hooks/useWatchPendingTransactions.ts

```diff
@@ -102,21 +102,30 @@ export const useWatchPendingTransactions = ({
         }
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
       } catch (e: any) {
+        const errorMessage = e?.message ?? String(e);
         logger.error(
           new RainbowError(
             `useWatchPendingTransaction: Failed to watch transaction`,
             { cause: e },
           ),
           {
-            message: e.message,
+            message: errorMessage,
             cause: e.cause,
             hash: tx.hash,
             chainId: tx.chainId,
           },
         );
+        if (
+          errorMessage.includes('Failed to create provider') ||
```

Comment:
e.code and an enum is nicer, but probably not worth the boilerplate; have become a bit Effect-pilled because of that